mood-analysis.rb

```diff
@@ -28,5 +36,51 @@ def analyze_mood(words)
   "05/11 Yay, yay, yay! I'm having a awfuly great day."
 ]
 
-puts analyze_mood(text[0])
-puts analyze_mood(text[1])
+sad_text = [
+  "03/01 I'm having a terrible horrible no good day.",
+  "03/13 Yesterday was horrible, but today is horrible!",
+  "04/02 Sad Panda. #terribleday",
+  "04/15 Hello World, today is horrible! ",
+  "05/01 Great! Yay! Good! Yay! Happy. Happy.",
+  "05/11 Yay, yay, yay! I'm having a awfuly great day."
+]
+
+def happy_days(entries)
+  happy = 0
+  no_of_days = 0
+    entries.each do |day|
+      no_of_days += 1
+      mood = analyze_mood(day)
+      if mood == ":-)"
+        happy += 1
+      end
+      break if happy == 3
+    end
+  print "It takes #{no_of_days} days for 3 happy days to occur."
+end
+
+def overall_mood(entries)
+  happy = []
+  sad = []
+  no_of_days = 0
+
+  entries.each do |day|
+    no_of_days += 1
+    mood = analyze_mood(day)
+    if mood == ":-)"
+      happy << day[0..4]
+    else sad << day[0..4]
+    end
+  end
```

Comment:
Oooh. This is interesting. I like that you stored the days that each mood happened on in separate arrays. Seems like maybe we could utilize this information for something cool, except that the array is stored in this method alone and not returned to anywhere else. I'd argue that a hash is a better choice here because all we are utilizing is emoticons and their counts (key/values). This is actually a common thing to see in whiteboarding interviews or problem solving of that nature (counting an undetermined number of things).